src/WebJobs.Extensions.SendGrid/Bindings/SendGridHelpers.cs

```diff
@@ -33,110 +35,80 @@ internal static bool TryParseAddress(string value, out MailAddress mailAddress)
+                    personalization.AddTo(config.ToAddress);
+                    mail.AddPersonalization(personalization);
                 }
             }
 
-            if (string.IsNullOrEmpty(message.Subject) &&
+            if (string.IsNullOrEmpty(mail.Subject) &&
                 !string.IsNullOrEmpty(attribute.Subject))
             {
-                message.Subject = attribute.Subject;
+                mail.Subject = attribute.Subject;
             }
 
-            if (string.IsNullOrEmpty(message.Text) &&
+            if ((mail.Contents == null || mail.Contents.Count == 0) &&
                 !string.IsNullOrEmpty(attribute.Text))
             {
-                message.Text = attribute.Text;
+                mail.AddContent(new Content("text/plain", attribute.Text));
             }
         }
 
-        internal static SendGridMessage CreateMessage(JObject input)
+        internal static Mail CreateMessage(string input)
         {
-            SendGridMessage message = new SendGridMessage();
-
-            JToken value = null;
-            MailAddress mailAddress = null;
-            if (input.TryGetValue("to", StringComparison.OrdinalIgnoreCase, out value))
-            {
-                Collection<MailAddress> addresses = new Collection<MailAddress>();
-                if (value.Type == JTokenType.Array)
-                {
-                    foreach (string address in value)
-                    {
-                        MailAddress to = null;
-                        if (!TryParseAddress(address, out to))
-                        {
-                            throw new ArgumentException("Invalid 'To' address specified");
-                        }
-                        addresses.Add(to);
-                    }
-                }
-                else if (value.Type == JTokenType.String)
-                {
-                    if (!TryParseAddress((string)value, out mailAddress))
-                    {
-                        throw new ArgumentException("Invalid 'To' address specified");
-                    }
-                    addresses.Add(mailAddress);
-                }
-
-                message.To = addresses.ToArray();
-            }
-
-            if (input.TryGetValue("from", StringComparison.OrdinalIgnoreCase, out value))
-            {
-                MailAddress from = null;
-                if (!TryParseAddress((string)value, out from))
-                {
-                    throw new ArgumentException("Invalid 'From' address specified");
-                }
-                message.From = from;
-            }
+            JObject json = JObject.Parse(input);
+            return CreateMessage(json);
+        }
 
-            if (input.TryGetValue("subject", StringComparison.OrdinalIgnoreCase, out value))
-            {
-                message.Subject = (string)value;
-            }
+        internal static Mail CreateMessage(JObject input)
+        {
+            return input.ToObject<Mail>();
```

Comment:
We were basically doing our own JSON deserialziation previously so we'd throw an error if something didn't parse as we expected. But now the `Mail` object is a lot more complex -- https://sendgrid.com/docs/API_Reference/Web_API_v3/Mail/index.html#-Request-Body-Parameters and the object supports JSON.NET properties: https://github.com/sendgrid/sendgrid-csharp/blob/master/SendGrid/SendGrid/Helpers/Mail/Mail.cs#L46. So I decided to remove all of that and just let JSON.NET handle the deserialization. The errors won't be as concise but I'm not sure we want to write code to handle all of the possibilities here.